contracts/protocol/pool/PoolCore.sol

```diff
@@ -253,6 +231,35 @@ contract PoolCore is
             );
     }
 
+    function decreaseUniswapV3Liquidity(
```

Comment:
we should probably make a new Pool contract and call it PoolUniswapV3.sol and add all the uniswap selectors to it.